bin.src/make_flats.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+"""
+Script to create imSim flats.  This includes electorstatic sensor effects
+such as treerings and brighter-fatter.
+"""
+import argparse
+import galsim
+import lsst.afw.cameraGeom as cameraGeom
+from lsst.sims.GalSimInterface import LSSTCameraWrapper, make_galsim_detector
+import desc.imsim
+
+parser = argparse.ArgumentParser(description="imSim flat production script.")
+parser.add_argument('instcat', type=str, help="instance catalog file")
+parser.add_argument('--sensors', type=str, default=None,
+                    help="sensors to simulate")
+parser.add_argument('--seed', type=int, default=42, help="random seed")
+parser.add_argument('--counts_total', type=int, default=8e4,
+                    help="total # electrons/pixel in flat")
+parser.add_argument('--counts_per_iter', type=int, default=4e3,
+                    help="# electrons/pixel per iteration")
+parser.add_argument('--log_level', type=str,
+                    choices=['DEBUG', 'INFO', 'WARN', 'ERROR', 'CRITICAL'],
+                    default='INFO', help='Logging level. Default: INFO')
+
+args = parser.parse_args()
+
+config = desc.imsim.read_config()
+
+obs_md, phot_params, _ = desc.imsim.parsePhoSimInstanceFile(args.instcat,
```

Comment:
This code is doing essentially the same calculation as `FastSiliconSkyModel` where a specific SED isn't used either.   As far as required instance catalog contents are concerned, this code just needs a complete set of phosim commands so that the output files have the same header information.  This is a requirement to make the `raw` files that would be ingested into the repo using `obs_lsstCam`.   I'd prefer to put all of that sort of documentation in the wiki, since there is a procedure associated with making the flats and other calibration products, so it would be better to have it all in one place.

Reply:
For a flat, I don't think the SED matters at all.  At least if we're ignoring fringing and/or assuming an SED without spectral lines.  Aside from that the only ways that the SED can affect anything are:

1. Setting the overall level from the integral of SED(lambda) * throughput(lambda) dlambda.  But I'm sure we don't actually care about getting that right for some physical SED and exposure time.  Rather, we are just targeting an overall level in the final image.

2. There is a slight effect in terms of the diffusion, since the different wavelengths convert at different levels in the silicon, so they will have different effective diffusion lengths.  However (a) we are already ignoring diffusion entirely in the approximation we are doing to make this fast and (b) diffusion doesn't matter anyway for a uniform input surface brightness, just spreading that around into a new uniform surface brightness with the electrons in different locations.

So I think it's completely correct to ignore the SED here.

Reply:
> I'd prefer to put all of that sort of documentation in the wiki, since there is a procedure associated with making the flats and other calibration products, so it would be better to have it all in one place.

OK that makes sense.  You mean the DM processing wiki as opposed to imSim documentation here is that right? It will  probably be good for us to also go back and document this in the imSim area too.